sdk/metric/doc.go

```diff
@@ -12,36 +12,107 @@
-// is being run on. That way when multiple instances of the code are collected
-// at a single endpoint their origin is decipherable.
-//
-// See [go.opentelemetry.io/otel/metric] for more information about
-// the metric API.
+/*
+Package metric provides an implementation of the OpenTelemetry metrics SDK.
+
+See https://opentelemetry.io/docs/concepts/signals/metrics/ for information
+about the concept of OpenTelemetry metrics and
+https://opentelemetry.io/docs/concepts/components/ for more information
+about OpenTelemetry SDKs.
+
+The entry point for the metric package is the MeterProvider. It is the
+object that all API calls use to create Meters, instruments, and ultimately
+make metric measurements. Also, it is an object that should be used to
+control the life-cycle (start, flush, and shutdown) of the SDK.
+
+A MeterProvider needs to be configured to export the measured data, this is
+done by configuring it with a Reader implementation (using the WithReader
+MeterProviderOption). Readers take two forms: ones that push to an endpoint
+(NewPeriodicReader), and ones that an endpoint pulls from. See
+[go.opentelemetry.io/otel/exporters] for exporters that can be used as
+or with these Readers.
+
+Each Reader, when registered with the MeterProvider, can be augmented with a
+View. Views allow users that run OpenTelemetry instrumented code to modify
+the generated data of that instrumentation.
+
+The data generated by a MeterProvider needs to include information about its
+origin. A MeterProvider needs to be configured with a Resource, using the
+WithResource MeterProviderOption, to include this information. This Resource
+should be used to describe the unique runtime environment instrumented code
+is being run on. That way when multiple instances of the code are collected
+at a single endpoint their origin is decipherable.
+
+See [go.opentelemetry.io/otel/metric] for more information about
+the metric API.
+
+# Experimental
+
+The metric SDK contains features that have not yet stabilized in the
+OpenTelemetry specification. These features are added to the OpenTelemetry Go
+metric SDK prior to stabilization in the specification so that users can start
+experimenting with them and provide feedback.
+
+These feature may change in backwards incompatible ways as feedback is applied.
+
+Experimental features do not fall within the scope of the OpenTelemetry Go
+versioning and stability policy. These features may be removed or modified in
+successive version releases, including patch versions.
+
+When an experimental feature is promoted to a stable feature, a migration path
+will be included in the changelog entry of the release. There is no guarantee
+that any environment variable feature flags that enabled the experimental
+feature will be supported by the stable version. If they are supported, they
+may be accompanied with a deprecation notice stating a timeline for the removal
+of that support.
+
+# Cardinality Limit (experimental)
```

Comment:
The lack of nest section in Go docs is limiting in a way that does not copy the original structure of the markdown file this is replacing.

Reply:
Add an experimental section that links to the exiting doc. That way the existing structure is preserved.

Reply:
I think it is impossible to add a relative hyperlink in Go doc.

I do not want to add a hyperlink to https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/metric/EXPERIMENTAL.md as it will refer to `main` (the docs will be changing).

I do not want to add a hyplink to a tag e.g. https://github.com/open-telemetry/opentelemetry-go/blob/v1.23.0/sdk/metric/EXPERIMENTAL.md as it would be unpleasant to maintain. Also I think having everything in one page is more cohesive.

Reply:
This is also following our contribution guidelines: https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#documentation